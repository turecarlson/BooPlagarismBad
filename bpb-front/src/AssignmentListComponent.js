import React from 'react';
import './AssignmentListComponent.css';

function AssignmentListComponent() {
  return (
    <div className="App">
      AssignmentListComponent
    </div>
  );
}

export default AssignmentListComponent;
