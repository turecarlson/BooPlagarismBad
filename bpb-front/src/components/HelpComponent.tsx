import React from 'react';
import { RouteComponentProps } from 'react-router';
import { Link } from 'react-router-dom';

interface PropTypes extends RouteComponentProps {}

/**
 * The help component consists of a basic render page where a user can see descriptions of how to create a submission
 * and an assignment as well as how to compare two submissions.
 */
class HelpComponent extends React.Component <PropTypes, {}> {

  constructor(props : PropTypes) {
    super(props);
  }

  render() {
    return (
      <div className="container card help-panel mt-3">
        <div className="card-body">
          <h2>Help / How-To</h2>
          <ul className="list-group">

            <li className="list-group-item">  
              <h4>Create an Assignment</h4>
              <ol>
                <li>Go to <Link to="/">Assignments</Link></li>
                <li>Choose Create New Assignment</li>
                <li>Enter a name for your assignment</li>
                <li>Choose Create Assignment</li>
                <li>Bang! An assignment is created!</li>
              </ol>
            </li>

            <li className="list-group-item">
              <h4>Make a Submission to an Assignment</h4>
              <ol>
                <li>Go to <Link to="/">Assignments</Link></li>
                <li>Choose the Assignment you are submitting to</li>
                <li>Select Upload Submission</li>
                <li>Choose files to submit by dropping them into the upload area</li>
                <li>Select Upload Submission</li>
                <li>Yes! Your submission is created! (hopefully)</li>
              </ol>
            </li>

            <li className="list-group-item">
              <h4>Compare Submissions</h4>
              <ol>
                <li>Go to <Link to="/">Assignments</Link></li>
                <li>Choose the Assignment you wish to view</li>
                <li>Choose the Submissions you wish to compare</li>
                <li>Select Compare Submissions</li>
                <li>In the Comparison view, select any file you wish to view from either submission</li>
                <li>Review the Similarity Percentage to see if the files shown are suspiciously similar...</li>
                <li>Select any Match you wish to view in more detail!</li>
              </ol>
            </li>

          </ul>
        </div>
      </div>
    );
  }
}

export default HelpComponent;