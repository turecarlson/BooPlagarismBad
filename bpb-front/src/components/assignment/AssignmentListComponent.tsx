import React from 'react';
import AssignmentListCard from './AssignmentListCard'
import Assignment from '../../types/Assignment'
import {Link} from 'react-router-dom'
import {store} from '../../store'
import { readAssignments } from '../../actions/AssignmentAction';

interface PropsType {
  //assignments:Assignment[]
}

class AssignmentListComponent extends React.Component <PropsType, {assignments: Assignment[]}> {
  //({assignments = [{_id:"01", name:"Mikayla", submissionIds:[]}]}: {assignments: Assignment[]})
  constructor(props: PropsType) {
    super(props);
    this.state = {
      assignments : store.getState().AssignmentReducer.assignments
    };
  }
  
  componentDidMount() {
    console.log("Call readAssignments from mount");
    // store.dispatch(readAssignments().then(() =>)).then(() => {
    //   this.setState({
    //     assignments : store.getState().AssignmentReducer.assignments
    //   });
    // });    

    store.dispatch(readAssignments())
    this.setState({
      assignments : store.getState().AssignmentReducer.assignments
    })
  }


  render () {
    return (        
      <div className='assignment-list pl-2'>
        <Link to="/CreateAssignment" className="new-assignment-btn btn btn-outline-info ml-2">Create New Assignment</Link> 
          {
            console.log("Assignments in component", this.state.assignments)
          }       
          {
            this.state.assignments.length > 0 &&
              <ul className='nav'>
                {
                  this.state.assignments.map((assignment,index) => 
                      <li className='nav-item' key={index}><AssignmentListCard assignment={assignment} createAssignment={(arg: String) => null}/></li>
                  )
                }
              </ul>
          }
          {
            this.state.assignments.length <= 0 &&
              <h4>No assignments to display</h4>
          }        
      </div>
    );
  }
}

export default AssignmentListComponent
