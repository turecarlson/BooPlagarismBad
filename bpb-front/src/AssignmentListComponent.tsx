import React from 'react';
import './AssignmentListComponent.css';
import AssignmentListCard from './AssignmentListCard'
import Assignment from './types/Assignment'
import {Link} from 'react-router-dom'

function AssignmentListComponent({assignments}: {assignments: Assignment[]}) {
  return (
    <div className='assignment-list'>
      <Link to="#" className="new-assignment-btn">new</Link>
      <ul>
        {
          assignments.map((assignment,index) => 
              <li key={index}><AssignmentListCard assignment={assignment} createAssignment={assignment}/></li>
           )
        }
      </ul>
    </div>
  );
}

export default AssignmentListComponent
