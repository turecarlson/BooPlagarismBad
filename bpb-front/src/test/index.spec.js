import React from 'react';
import AssignmentListComponent from '../components/assignment/AssignmentListComponent';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router'
import HelpComponent from '../components/HelpComponent';


describe.skip('index.js', () => {
    it('should render AssignmentListComponent', () => {
        const component = renderer.create(
            <StaticRouter location='/' context={{}}>
                <AssignmentListComponent assignments={[]}/>
            </StaticRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render HelpComponent', () => {
        const component = renderer.create(
            <StaticRouter location='/HelpComponent' context={{}}>
                <HelpComponent/>
            </StaticRouter>
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

// add additional testing for onclicking nav bar