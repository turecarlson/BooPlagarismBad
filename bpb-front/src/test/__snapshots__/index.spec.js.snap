// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.js should render AssignmentListComponent 1`] = `
<div
  className="App"
>
  AssignmentListComponent
</div>
`;
